### PR TITLE
integration test for new ca cert in kubernetes tls secret

### PR DESCRIPTION
**Pull request by williamaronli**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: tests/integration/security/sds_ingress/util/util.go [hunk 1]
```diff
@@ -162,8 +164,9 @@ func createSecret(ingressType ingress.CallType, cn, ns string, ic IngressCredent
 			Namespace: ns,
 		},
 		Data: map[string][]byte{
-			tlsScrtCert: []byte(ic.ServerCert),
-			tlsScrtKey:  []byte(ic.PrivateKey),
+			tlsScrtCert:   []byte(ic.ServerCert),
+			tlsScrtKey:    []byte(ic.PrivateKey),
+			tlsScrtCaCert: []byte(ic.CaCert),
```
Comment (howardjohn) [inline tests/integration/security/sds_ingress/util/util.go:169]: This Doesn't seem right. `ingressType != ingress.Mtls` yet we are creating a mTLS secret?

For plain TLS we should not have a ca cert

File: tests/integration/security/sds_ingress/ingress_test.go [hunk 2]
```diff
@@ -172,7 +172,69 @@ func TestSingleMTLSGateway_CompoundSecretRotation(t *testing.T) {
...
+			}
+			err = ingressutil.SendRequest(ing, host, credName[0], ingress.Mtls, tlsContext,
+				ingressutil.ExpectedResponse{ResponseCode: 200, ErrorMessage: ""}, t)
+			if err != nil {
+				t.Fatalf("unable to retrieve 200 from product page at host %s: %v", host, err)
+			}
+		})
+}
+
+// TestSingleMTLSGateway_CompoundSecretRotation tests a single mTLS ingress gateway with SDS enabled and use the tls cert instead of generic cert
```
Comment (JimmyCYJ) [inline tests/integration/security/sds_ingress/ingress_test.go:220]: Update the comment? TestSingleMTLSGatewayWithCaCert_CompoundSecretRotation

File: tests/integration/security/sds_ingress/util/util.go [hunk 3]
```diff
@@ -88,7 +90,7 @@ var IngressCredentialServerKeyCertB = IngressCredential{
 // and creates K8s secrets for ingress gateway.
 // nolint: interfacer
 func CreateIngressKubeSecret(t test.Failer, ctx framework.TestContext, credNames []string,
-	ingressType ingress.CallType, ingressCred IngressCredential) {
+	ingressType ingress.CallType, ingressCred IngressCredential, isGenericSecret bool) {
```
Comment (JimmyCYJ) [inline tests/integration/security/sds_ingress/util/util.go:93]: Change isGenericSecret to isCompoundGeneric, to make it clear that this flag is only used for creating compound secret.

File: tests/integration/security/sds_ingress/ingress_test.go [hunk 4]
```diff
@@ -172,7 +172,69 @@ func TestSingleMTLSGateway_CompoundSecretRotation(t *testing.T) {
...
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			var (
+				credName = []string{"testsinglemtlsgatewaywithcacert-compoundsecretrotation"}
+				host     = "testsinglemtlsgatewaywithcacert-compoundsecretrotation.example.com"
+			)
+
+			// Add kubernetes secret to provision key/cert for ingress gateway.
+			ingressutil.CreateIngressKubeSecret(t, ctx, credName, ingress.Mtls, ingressutil.IngressCredentialA, true)
```
Comment (JimmyCYJ) [inline tests/integration/security/sds_ingress/ingress_test.go:237]: isCompoundGeneric should be false in this case, is it?

File: tests/integration/security/sds_ingress/ingress_test.go [hunk 5]
```diff
@@ -172,7 +172,69 @@ func TestSingleMTLSGateway_CompoundSecretRotation(t *testing.T) {
...
+				Cert:       ingressutil.TLSClientCertA,
+			}
+			err := ingressutil.SendRequest(ing, host, credName[0], ingress.Mtls, tlsContext,
+				ingressutil.ExpectedResponse{ResponseCode: 200, ErrorMessage: ""}, t)
+			if err != nil {
+				t.Fatalf("unable to retrieve 200 from product page at host %s: %v", host, err)
+			}
+
+			// key/cert rotation
+			ingressutil.RotateSecrets(t, ctx, credName, ingress.Mtls, ingressutil.IngressCredentialB)
```
Comment (JimmyCYJ) [inline tests/integration/security/sds_ingress/ingress_test.go:198]: RotateSecrets() needs an update as well.

File: tests/integration/security/sds_ingress/util/util.go (hunk 3 above)
Comment (williamaronli) [reply tests/integration/security/sds_ingress/util/util.go:93]: i think we should not set it specific for compound secret. As many other functions use this flag. How about I set a new flag called the isCompoundWithCaCert to specify for that test

File: tests/integration/security/sds_ingress/ingress_test.go (hunk 4 above)
Comment (williamaronli) [reply tests/integration/security/sds_ingress/ingress_test.go:237]: Yes it should be a false here. However as I change the flag from isCompoundGeneric to isCompoundWithCaCert to specify the integration test I added. I will leave it true. And all other function will use the parameter as false.

File: tests/integration/security/sds_ingress/util/util.go [hunk 6]
```diff
@@ -142,17 +144,31 @@ func DeleteIngressKubeSecret(t test.Failer, ctx framework.TestContext, credNames
...
 		return &v1.Secret{
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      cn,
 				Namespace: ns,
 			},
 			Data: map[string][]byte{
-				genericScrtCert:   []byte(ic.ServerCert),
-				genericScrtKey:    []byte(ic.PrivateKey),
-				genericScrtCaCert: []byte(ic.CaCert),
+				tlsScrtCert:   []byte(ic.ServerCert),
```
Comment (JimmyCYJ) [inline tests/integration/security/sds_ingress/util/util.go:169]: This part should not change.

File: tests/integration/security/sds_ingress/ingress_test.go [hunk 7]
```diff
@@ -195,7 +195,69 @@ func TestSingleMTLSGateway_CompoundSecretRotation(t *testing.T) {
...
+// termination is working properly. This secret is a compound secret.
+// (2) After key/cert rotation, client needs to pick new CA cert to complete SSL connection. Old CA
+// cert will cause the SSL connection fail.
+func TestSingleMTLSGatewayAndNotGeneric_CompoundSecretRotation(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			var (
+				credName = []string{"testsinglemtlsgatewaywithcacert-compoundsecretrotation"}
```
Comment (JimmyCYJ) [inline tests/integration/security/sds_ingress/ingress_test.go:232]: Please use testsinglemtlsgatewayandnotgeneric-compoundsecretrotation for credName and host.

**Pull request merged**